### ISSUE TITLE
Authentication and redirect/callback to custom UI running in parallel to kubeflow - Dex/OIDC

### ISSUE THREAD

**Issue by amitchatbd** [kubeflow/kubeflow#5656]
/kind question

**Question:**
Is there a way to redirect to a different domain after authentication and getting the JWT token.

This is the scenario, we are developing a UI integration providing kubeflow capabilities and other functions for a group of users. 
We need to authenticate and get the core JWT token for kubeflow which is configured to validate the user against internal LDAP in our current kubeflow env. Our custom app needs authenticate independently to get that JWT token and use it to call kubefow API's to allowing users to create notebook servers from out custom UI and do other kubeflow stuff through different kubeflow API.
We have a default Kubeflow environment with Dex.

In current config calling the dex URL provided by kubeflow redirects to native kubeflow install and URL. How to intercept authentication from our custom UI-URL and auth from our custom app URL and redirect or callback after getting JWT token from kubeflow default install.

Any API, Istio or Envoy intercept and redirect? 

Help is really appreciated.

Thanks

**Comment by DavidSpek** [comment]
Do you need to use Dex? You could use a different OIDC provider for kubeflow, which could be your custom app or something like GitLab.

**Comment by amitchatbd** [comment]
Our install is with OIDC and Dex. Using internal LDAP. We will not use any other options at this point.

Possibility of Ping federated integration in the future ..... that is not part of this question.

**Comment by ReggieCarey** [comment]
I face a similar situation.  We would like to authenticate and authorize access to things other than Kubeflow.  This is not specifically a Kubeflow issue but it seems that Kubeflow has solved a problem that we would like to solve for other products.

**Comment by amitchatbd** [comment]
For us we are trying to access kubeflow API's and integrate other functionality that we are developing with those calls so that we can provide a better interactive UI to specific user group in our org.

**Comment by DavidSpek** [comment]
@amitchatbd You can setup redirects with the Istio VietualService and do things like adding, changing or removing both request and response headers. Not sure if this is of any use for you though. https://istio.io/latest/docs/reference/config/networking/virtual-service/

**Comment by jbottum** [comment]
@amitchatbd you might scan some of the materials created by the Kubeflow team, https://journal.arrikto.com/kubeflow-authentication-with-istio-dex-5eafdfac4782, and this presentation: https://www.youtube.com/watch?v=U8yWOKOhzes
